### DISCUSSION TITLE
Asset File Extension In UrlReference

### DISCUSSION THREAD

**Discussion by Bodyclock-Games** [stride3d/stride#1902] [Ideas]
I'm loading a 16bit raw image  as a RawAsset through the Content.OpenAsStream method. All works great. But I want to do some checks to make sure the asset is a raw file. One quick check is through the file extension. I tried the following code to get the extension:

`RawAsset raw = Content.Load<RawAsset>(RawAssetUrl.Url);
string ext = raw.MainSource.GetFileExtension();
Log.Error(ext);
Content.Unload(raw); `

But I get the following error, which on reflection makes sense.
Stride.Core.Serialization.Contents.ContentManagerException: 'Unexpected exception while loading asset [Height Maps/Heights]. Reason: Content serializer for /Stride.Core.Assets.RawAsset could not be found.

I wondered if asset extension information could be added as a property to the UrlReference. It would be useful for a rudimentary check of the raw asset type.

**Comment by manio143** [comment]
I don't think it makes sense to add the file extension. The raw asset could be any binary data (possibly in your custom format with no extension).
Instead I would suggest that you follow some naming convention for you asset files thus having a meaningful URL.

For example you can add a suffix to your raw asset URl like`_{ext}` and then all you need to do is check `RawAssetUrl.Url.EndsWith("_smth")`.
